portal-api/src/v1/users/validation/rules/read-only-role-cannot-be-assigned-with-other-roles.test.js

```diff
@@ -0,0 +1,68 @@
+const readOnlyRoleCannotBeAssignedWithOtherRoles = require('./read-only-role-cannot-be-assigned-with-other-roles');
+const { READ_ONLY, ALL_ROLES } = require('../../../roles/roles');
+
+describe('readOnlyRoleCannotBeAssignedWithOtherRoles', () => {
+  const user = {};
+  const nonReadOnlyRoles = ALL_ROLES.filter((role) => role !== READ_ONLY);
+  const allPairsOfNonReadOnlyRoles = nonReadOnlyRoles.flatMap(
+    (leftRole) => nonReadOnlyRoles.map(
+      (rightRole) => ({ leftRole, rightRole })
+    )
+  );
```

Comment:
Love this!

We're repeating a fair amount of items here... ie when we have 4 roles, well have an array of 16, when in fact there are either only 10 unique combinations if we are not bothered by the order. If we are not bothered by dups (ie `[read-only,read-only]`) this goes down to 6.

To include dupes (so we get 10 test cases)

```suggestion
const allPairsOfNonReadOnlyRoles = nonReadOnlyRoles.flatMap((leftRole, i) => {
    const rightRoleList = [...nonReadOnlyRoles].slice(i);
  return rightRoleList.map((rightRole) => ({leftRole, rightRole}))
})
```

To exclude dupes (so we get 6 test cases):
```suggestion
const allPairsOfNonReadOnlyRoles = nonReadOnlyRoles.flatMap((leftRole, i) => {
    const rightRoleList = [...nonReadOnlyRoles].slice(i+1);
  return rightRoleList.map((rightRole) => ({leftRole, rightRole}))
})
```

Reply:
Happy days! :)